vector/src/main/java/im/vector/app/features/lifecycle/VectorActivityLifecycleCallbacks.kt

```diff
@@ -42,5 +58,81 @@ class VectorActivityLifecycleCallbacks constructor(private val popupAlertManager
     }
 
     override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {
+        // restart the app if the task contains an unknown activity
+        if (isTaskCorrupted(activity)) {
```

Comment:
would be interesting to know if there's a performance impact for this, might be worth making the check async to avoid slowing down the activity start flow

Reply:
Done